src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -90,66 +88,52 @@ private void AnalyzeArgument(OperationAnalysisContext context)
             }
         }
 
-        private static IEnumerable<string> GetPropertiesInScope(OperationAnalysisContext context)
+        private static bool HasAPropertyMatchInScope(OperationAnalysisContext context, string stringText)
         {
             var containingType = context.ContainingSymbol.ContainingType;
-            // look for all of the properties in the containing type and return the property names
-            if (containingType != null)
-            {
-                foreach (var property in containingType.GetMembers().OfType<IPropertySymbol>())
-                {
-                    yield return property.Name;
-                }
-            }
+            return containingType?.GetMembers(stringText).Any(m => m.Kind == SymbolKind.Property) == true;
         }
 
-        internal static IEnumerable<string> GetParametersInScope(OperationAnalysisContext context)
+        private static bool HasAParameterMatchInScope(OperationAnalysisContext context, string stringText)
         {
             // get the parameters for the containing method
             foreach (var parameter in context.ContainingSymbol.GetParameters())
             {
-                yield return parameter.Name;
+                if (parameter.Name == stringText)
+                {
+                    return true;
+                }
             }
 
             // and loop through the ancestors to find parameters of anonymous functions and local functions
             var parentOperation = context.Operation.Parent;
             while (parentOperation != null)
             {
+                IMethodSymbol? methodSymbol = null;
```

Comment:
IMO it would be clearer to use a switch expression on the rhs of the assignment with the default case being `null`.